### PR TITLE
feat: add default_headers to Gemini model

### PR DESCRIPTION
**Pull request by ashishb**
# Description

Similar to [OpenAI](https://github.com/run-llama/llama_index/blob/9900560183a2e212515e6273e63159316e056e96/llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py#L190-L192), allow `default_headers` for Gemini model as well

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it made sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by ashishb** [comment]
@logan-markewich can you please review?

File: llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/base.py [hunk 1]
```diff
@@ -125,6 +126,11 @@ def __init__(
             config_params["client_options"] = {"api_endpoint": api_base}
         if transport:
             config_params["transport"] = transport
+        if default_headers:
+            default_metadata: Sequence[Dict[str, str]] = []
+            for key, value in default_headers.items():
+                default_metadata.append((key, value))
+            config_params["default_metadata"] = default_metadata
```
Comment (masci) [inline llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/base.py:133]: I'm not a `genai` expert and it took me a while to figure out why something called `default_metadata` would end up becoming HTTP headers 😅 so I thought a comment would help here.

```suggestion
            # `default_metadata` contains (key, value) pairs that will be sent with every request.
            # When using `transport="rest"`, these will be sent as HTTP headers.
            config_params["default_metadata"] = default_metadata
```

**Review by masci — CHANGES_REQUESTED** [review]
Left a comment about documentation.

**Comment by ashishb** [comment]
> Left a comment about documentation.

Thanks. Accepted.

**Review by masci — APPROVED** [review]

**Comment by masci** [comment]
Bumped the version so integration will be released on merge.

**Pull request merged**